core/domain/html_validation_service.py

```diff
@@ -0,0 +1,890 @@
+        err_msg = "Invalid attributes"
+        return err_msg
+
+    for customization_arg in customization_args:
+        arg_name = customization_arg['name'] + '-with-value'
+        arg_value = json.loads(unescape_html(attrs[arg_name]))
+        schema = customization_arg['schema']
+        if tag_name != 'oppia-noninteractive-tabs':
+            if schema['type'] != 'custom':
+                arg_type = obj_types_to_obj_classes[schema['type']]
+            else:
+                arg_type = obj_types_to_obj_classes[schema['obj_type']]
+            try:
+                arg_type.normalize(arg_value)
+                # To ensure filepath is in valid format.
+                if tag_name == 'oppia-noninteractive-image':
+                    if arg_type == objects.Filepath:
+                        if not re.match(filename_re, arg_value):
+                            err_msg = "Invalid filepath"
+
+                # To ensure video id has 11 chars.
+                elif tag_name == 'oppia-noninteractive-video':
+                    if arg_type == objects.UnicodeString:
+                        if len(arg_value) != 11:
+                            err_msg = "Video id length is not 11"
+
+                # To ensure that nested collapsible or tab
+                # is not present in collapsible component.
+                elif tag_name == 'oppia-noninteractive-collapsible':
+                    inner_soup = bs4.BeautifulSoup(
+                        arg_value.encode(encoding='utf-8'),
+                        'html.parser')
+                    collapsible_count = len(inner_soup.findAll(
+                        name='oppia-noninteractive-collapsible'))
+                    tabs_count = len(inner_soup.findAll(
+                        name='oppia-noninteractive-tabs'))
+                    if collapsible_count or tabs_count:
+                        err_msg = "Nested tabs and collapsible"
+
+            except Exception as e:
+                err_msg = str(e)
+        else:
+            # This block is to check args for tab component.
+            # Ensure that arg is a list and each element item in
+            # the list is a dictionary.
+            if isinstance(arg_value, list) and (
+                    all(isinstance(item, dict) for item in (
+                        arg_value))):
+                for value in arg_value:
+                    value_keys = value.keys()
+                    # Each item has exactly two keys namely content
+                    # and title.
+                    if set(value_keys) != set(['content', 'title']):
+                        err_msg = "Invalid attributes in tab contents"
+                    else:
+                        content = value['content']
+                        title = value['title']
+                        title_dict = schema['items']['properties'][0]
+                        content_dict = schema['items']['properties'][1]
+                        title_type = obj_types_to_obj_classes[
```

Comment:
A type is a string. title_obj_class is what you want.

Ditto below.

Reply:
Done!